core/Decode.cpp

```diff
@@ -112,6 +123,20 @@ namespace olympia
         }
     }
 
+    // process vset settings being forward from execution pipe
+    // for set instructions that depend on register
+    void Decode::process_vset_(const InstPtr & inst)
+    {
+        VCSRs_.lmul = inst->getLMUL();
+        VCSRs_.sew = inst->getSEW();
+        VCSRs_.vl = inst->getVL();
+        VCSRs_.vta = inst->getVTA();
+        VCSRs_.vlmax = (Inst::VLEN / VCSRs_.sew) * VCSRs_.lmul;
```

Comment:
Does `vlmax` get initialized properly at the start of simulation with the initial values?

Reply:
Just answered my own question by looking at `Inst::VCSRs` struct. I would add a method to this struct so the vlmax calculation isn't defined in two places.

Reply:
Addressed in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)